### PR TITLE
Ruby: fix all QL-QL alerts

### PR DESCRIPTION
**Pull request by aibaars**
Fixes all alerts reported in:

`gh api -H 'Accept: application/sarif+json' https://api.github.com/repos/github/codeql/code-scanning/analyses/20490376`

https://github.com/github/codeql/pull/7814/checks?check_run_id=5036474517

### CODE REVIEW

**Review by aschackmull — APPROVED** [review]

**Review by nickrolfe — COMMENTED** [review]
The changes look good, but I'm not familiar with how `tutorial.qll` is used. Is there some associated document that might be affected?

**Review by erik-krogh — DISMISSED** [review]
:rocket:

File: ruby/ql/lib/codeql/ruby/ast/internal/Variable.qll [hunk 1]
```diff
@@ -414,10 +414,10 @@ class TVariableReal =
 class TLocalVariable = TLocalVariableReal or TLocalVariableSynth or TSelfVariable;
 
 /**
- * This class only exists to avoid negative recursion warnings. Ideally,
- * we would use `VariableImpl` directly, but that results in incorrect
- * negative recursion warnings. Adding new root-defs for the predicates
- * below works around this.
+ * A "real" (ie. non-synthesized) variable. This class only exists to
```
Comment (hvitved) [inline ruby/ql/lib/codeql/ruby/ast/internal/Variable.qll:417]: i.e.

**Review by MathiasVP — APPROVED** [review]

**Comment by aibaars** [comment]
> The changes look good, but I'm not familiar with how `tutorial.qll` is used. Is there some associated document that might be affected?

I don't think so. The tutorial is mentioned here https://codeql.github.com/docs/writing-codeql-queries/crown-the-rightful-heir/ .

**Review by nickrolfe — APPROVED** [review]

**Review by erik-krogh — APPROVED** [review]

**Review by tausbn — APPROVED** [review]

**Review by geoffw0 — COMMENTED** [review]
CPP :+1:

**Review by hvitved — APPROVED** [review]

**Pull request merged**